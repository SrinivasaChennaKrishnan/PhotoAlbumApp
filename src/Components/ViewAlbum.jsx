import React from "react";
import Classes from "./Styles/CommonStyle.module.css";
import * as Layout from "react-bootstrap";
import * as Action from "../Actions/AlbumAction";
import { store } from "../AppStore/store";
import * as ActionType from "../Common/ActionTypes";

/* Called on click of Back to album link */
const backToAlbum = () => {
  store.dispatch(Action.backToAlbumAction());
};

/* Called on click of Back to Photo from Album */
const showPhoto = photoObj => {
  store.dispatch({
    type: ActionType.SHOW_LOADER,
    showLoader: true
  });
  store.dispatch(Action.showPhotoAction(photoObj));
};

/* View Album Component */
function ViewAlbum(props) {
  let photolistObject =
    ((props.propData || {}).albumlist || {}).photolistObject || [];
  let selectedAlbum =
    ((props.propData || {}).albumlist || {}).selectedAlbum || [];
  let albumTitle = (selectedAlbum || []).title || null;
  return (
    <div className={Classes.center}>
      <Layout.Container className={Classes.containerOutline}>
        {/* Link  to Navigate back to Album*/}
        <Layout.Row className={Classes.backToAlbumRow}>
          <p
            className={Classes.backToAlbum}
            onClick={() => {
              backToAlbum();
            }}
          >
            {" << Back to Albums"}
          </p>
        </Layout.Row>
        <Layout.Row className={Classes.photoHeadRow}>
          <p className={Classes.albumHeading}>Photo Gallery</p>
        </Layout.Row>

        {/* Showing the Album & photos information */}
        <Layout.Row>
          <p className={Classes.ml20}>
            <span className={Classes.albumDescription}>
              Showing{" "}
              <span className={Classes.bold}>{photolistObject.length}</span>{" "}
              Photos from the Album{" "}
            </span>{" "}
            <span className={Classes.albumDescriptionAlbumName}>
              {albumTitle}
            </span>
          </p>
        </Layout.Row>

        {/* Listing the Photos from the respective Album */}
        <Layout.Container className={Classes.albumOutline}>
          <Layout.Row>
            {photolistObject &&
              photolistObject.length > 0 &&
              photolistObject.map((photo, index) => {
                return (
                  <Layout.Col xs lg="2" key={index}>
                    <div
                      key={index}
                      className={Classes.hvrGrow}
                      onClick={() => {
                        showPhoto(photo);
                      }}
                    >
                      <img
                        src={photo.thumbnailUrl}
                        alt={"album" + photo.id}
                        className={Classes.albumImage}
                      />
                      <p className={Classes.albumTitle}>{photo.title}</p>
                    </div>
                  </Layout.Col>
                );
              })}
          </Layout.Row>
        </Layout.Container>
      </Layout.Container>
    </div>
  );
}

export default ViewAlbum;
