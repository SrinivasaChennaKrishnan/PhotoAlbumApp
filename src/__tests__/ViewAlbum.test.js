import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ViewAlbum from '../Components/ViewAlbum'

/** Test View Album Component renders without crashing  */
describe('Album', () => {
    it('renders without crashing given the required props', () => {
        const props = {
            albumData: [],
            photolistObject: [],
            loader: false,
            albumMessage: "INITIAL",
            photoListMessage: "INITIAL",
            picSource: "",
            showPhoto: false,
            selectedAlbum: [],
            selectedPhoto: [],
            dispatch: jest.fn()
        }
        const wrapper = shallow(<ViewAlbum {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
