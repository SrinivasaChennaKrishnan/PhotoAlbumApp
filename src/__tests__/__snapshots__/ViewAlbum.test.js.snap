// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album renders without crashing given the required props 1`] = `
<div
  className="center"
>
  <Container
    className="containerOutline"
    fluid={false}
  >
    <ForwardRef
      className="backToAlbumRow"
      noGutters={false}
    >
      <p
        className="backToAlbum"
        onClick={[Function]}
      >
         &lt;&lt; Back to Albums
      </p>
    </ForwardRef>
    <ForwardRef
      className="photoHeadRow"
      noGutters={false}
    >
      <p
        className="albumHeading"
      >
        Photo Gallery
      </p>
    </ForwardRef>
    <ForwardRef
      noGutters={false}
    >
      <p
        className="ml20"
      >
        <span
          className="albumDescription"
        >
          Showing
           
          <span
            className="bold"
          >
            0
          </span>
           
          Photos from the Album
           
        </span>
         
        <span
          className="albumDescriptionAlbumName"
        />
      </p>
    </ForwardRef>
    <Container
      className="albumOutline"
      fluid={false}
    >
      <ForwardRef
        noGutters={false}
      />
    </Container>
  </Container>
</div>
`;
